### ISSUE TITLE
false-positive: assign dereferenced pointer struct

### ISSUE THREAD

**Issue by ldez** [sivchari/nilassign#5]
```go
package main

import "fmt"

func main() {
	f := &Foo{Name: "foo"}
	*f = Foo{Name: "bar"}

	fmt.Println(f)
}

type Foo struct {
	Name string
}
```

```console
$ ./golangci-lint run
main.go:7:2: this assignment occurs invalid memory address or nil pointer dereference (nilassign)
        *f = Foo{Name: "bar"}
        ^
```

**Comment by sivchari** [comment]
this problem is fixed. thank you.

**Comment by ldez** [comment]
can you point the commit of the fix?

**Comment by ldez** [comment]
Fixed by #6
